src/batch_decoder.cpp

```diff
@@ -32,8 +32,8 @@ batch_decoder::batch_decoder(shared_ptr<batch_iterator>                 b_itor,
     , m_provider(prov)
     , m_deterministic_mode(seed != 0)
 {
-    m_thread_pool =
-        singleton<thread_pool_queue<batch_decoder, &batch_decoder::process>>::get(thread_count);
+    m_thread_pool = singleton<thread_pool_queue<batch_decoder, &batch_decoder::process>>::get(
+        thread_affinity_map);
     m_number_elements_in = prov->get_input_count();
```

Comment:
I propose to pass thread_affinity_map by value , line 26
and add std::move() here

Reply:
done